File: challenges/2026-08-23-capped-team-merges/solution.py
from typing import List


def capped_team_merges(n: int, max_size: int, requests: List[List[int]]) -> List[int]:
    """Return the final team size of every employee after processing requests."""
    # TODO: implement
    pass


if __name__ == "__main__":
    tests = [
        (5, 3, [[0, 1], [2, 3], [1, 2], [3, 4]], [2, 2, 3, 3, 3]),
        (4, 2, [[0, 1], [1, 2], [2, 3]], [2, 2, 2, 2]),
        (3, 1, [[0, 1], [1, 2]], [1, 1, 1]),
        (1, 1, [], [1]),                                   # extra: single employee, no requests
        (4, 4, [[0, 1], [2, 3], [1, 2]], [4, 4, 4, 4]),     # extra: cap allows full merge
    ]

    for n, max_size, requests, expected in tests:
        got = capped_team_merges(n, max_size, requests)
        status = "PASS" if got == expected else "FAIL"
        print(f"{status} n={n} maxSize={max_size} requests={requests} expected={expected} got={got}")

File: challenges/2026-08-23-capped-team-merges/solution.go
package main

import "fmt"

func cappedTeamMerges(n int, maxSize int, requests [][]int) []int {
	// TODO: implement
	return nil
}

type testCase struct {
	n        int
	maxSize  int
	requests [][]int
	expected []int
}

func slicesEqual(a, b []int) bool {
	if len(a) != len(b) {
		return false
	}
	for i := range a {
		if a[i] != b[i] {
			return false
		}
	}
	return true
}

func main() {
	tests := []testCase{
		{5, 3, [][]int{{0, 1}, {2, 3}, {1, 2}, {3, 4}}, []int{2, 2, 3, 3, 3}},
		{4, 2, [][]int{{0, 1}, {1, 2}, {2, 3}}, []int{2, 2, 2, 2}},
		{3, 1, [][]int{{0, 1}, {1, 2}}, []int{1, 1, 1}},
		{1, 1, [][]int{}, []int{1}},                                     // extra: single employee, no requests
		{4, 4, [][]int{{0, 1}, {2, 3}, {1, 2}}, []int{4, 4, 4, 4}}, // extra: cap allows full merge
	}

	for _, tc := range tests {
		got := cappedTeamMerges(tc.n, tc.maxSize, tc.requests)
		status := "FAIL"
		if slicesEqual(got, tc.expected) {
			status = "PASS"
		}
		fmt.Printf("%s n=%d maxSize=%d requests=%v expected=%v got=%v\n", status, tc.n, tc.maxSize, tc.requests, tc.expected, got)
	}
}
